### ISSUE TITLE
[Bug] Invalid guessAtPath crashes babel plugin

### ISSUE THREAD

**Issue by jmeistrich** [marklawlor/tailwindcss-react-native#25]
We have some aliased paths in our tsconfig.json for sharing code between web/RN, so we import common code like:

```js
import { Info } from 'common/Interfaces'
```

This import causes the babel plugin to crash on `lstatSync` in `get-import-blocked-components.js` because the `guessAtPath` is not respecting the paths defined in tsconfig.json, so it's an invalid directory. 

These lines in tsconfig.json set up that path alias:
```json
{
    "compilerOptions": {
        "paths": {
            "common/*": [
                "../common/*"
            ]
        }
    }
}
```

For our usage we're using `allowModules` so we can just patch these lines out for now.

**Comment by marklawlor** [comment]
This is a little bit poorly documented, but you'll need to add `allowModules: ["common/*"]` (or just `common/Interfaces` if you don't want to match everything in common) to your options to match all imports from `common`

I'll add it to the troubleshoot section in the readme.

**Comment by marklawlor** [comment]
An error message has been added in `0.0.28` which should help guide people in the right direction

**Comment by jmeistrich** [comment]
But I don't want to allow those modules. I have set my allowModules to this:
```js
allowModules: ['./src-rn/**/*.tsx']
```

And that should be all the configuration I need to match only files in the `src-rn` folder? It seems like it would be quite cumbersome to require specifically allowing or blocking all possible paths and aliases?

**Comment by marklawlor** [comment]
If you don't want those modules processed then you can add them to `blockModules`

```
blockModules: ["common/*"]
```

**Comment by jmeistrich** [comment]
It does not work with file extensions. I see that the tests are importing from a directory. But I hit the error if I import from a file, because guessAtPath doesn't try joining any extensions.

For example, importing from `TestComponent.tsx`:
```jsx
import { TestComponent } from 'rn/TestComponent';
```

I did a quick hack to try `guessAtPath + '.tsx'` if it fails, and it stops crashing. But then the `blockList` in `web-visitor` includes `View` and `Text` which seems to basically block everything?

This is what's in my `babel.config.js`:
```
'tailwindcss-react-native/babel',
{
    platform: 'web',
    allowModules: ['rn/**/*.tsx'],
    blockModules: ['common/*'],
}
```

**Comment by marklawlor** [comment]
Can you try

```
{
    platform: 'web',
    allowModules: ['rn/*'],
    blockModules: ['common/*'],
}
```

**Comment by jmeistrich** [comment]
I can't seem to add enough paths to `blockModules` to make it stop crashing because it's not checking for file extensions. It got stuck crashing on an import from `./src/Components/DivEnsureInView` (a reactjs web component) and I can't figure out a glob to make it not crash. 

But if I leave in the hack to also try appending `.ts` and `.tsx` to `guessAtPath`, then it doesn't crash but seems to have blocked everything and renders no styles.

This is what I'm trying now:
```js
'tailwindcss-react-native/babel',
{
    platform: 'web',
    allowModules: ['rn/*'],
    blockModules: ['common/**/*', '~/**/*', 'src/**/*', './src/**/*'],
},
```

I believe in an ideal world my setup would look like this:

tailwind.config.js:
```js
module.exports = {
    content: ['./pages/**/*.{js,ts,jsx,tsx}', './src/**/*.{js,ts,jsx,tsx}', './src-rn/**/*.{js,ts,jsx,tsx}'],
    // ...
}
```

babel.config.js:
```js
'tailwindcss-react-native/babel',
{
    platform: 'web',
    allowModules: ['rn/*'], // aliased to ./src-rn
    // blockModules not needed because allowModules is enough
},
```

And FYI my hack to make it work for us is patching this into `getAllowedPaths` and disabling the `blockList` in `web-visitor`, which seems to make everything work perfectly:
```js
return {
    allowModules,
    allowRelativeModules: allowModules,
};
```

**Comment by marklawlor** [comment]
The current config you are using is still incorrect.

```
'tailwindcss-react-native/babel',
{
    platform: 'web',
    allowModules: ['rn/*'],
    blockModules: ['common/**/*', '~/**/*', 'src/**/*', './src/**/*'],
},
```

`allowModules` and `blockModules` only accept's names of modules - not file paths.

Do you have an example repo that I can test with?

I have a unit test [allow-modules-with-content](https://github.com/marklawlor/tailwindcss-react-native/tree/main/__tests__/native-inline/allow-modules-with-content) that I believe follows a similar scenario as the one you are describing.

```
// babel options
{
  "allowModules": ["react-native"],
}

// tailwind options.
module.exports = {
  content: [`${__dirname}/code.{js,ts,jsx,tsx}`],
};
```

It successfully only transforms the `Text` component from `react-native` and does nto transform the `moti` or `TestComponent`

```
import { Text } from "react-native";
import { TailwindProvider } from "tailwindcss-react-native";
import { MotiText } from "moti";
import {TestComponent} from "./test";

export function Test() {
  return (
    <TailwindProvider>
      <Text className="font-bold">Hello world!</Text>
      <MotiText className="font-bold">Do not transform</MotiText>
      <TestComponent className="font-bold">Do not transform</TestComponent>
    </TailwindProvider>
  );
}
```

**Comment by jmeistrich** [comment]
As I was making a repo I found I'm hitting two different issues:
1. I made a repo of the general concept of mixing web with react-native-web and it seems that anything in the tailwind config's content is always imported because it's in `allowRelativeModules`, and it never checks `allowModules` / `blockModules`. I added a new test `web/mix-web-rn` in a fork: https://github.com/jmeistrich/tailwindcss-react-native/tree/test-mix-web-rn
2. TypeScript aliases seem to cause problems when they're a couple of levels deep. For example, `blockModules: ["~/*"]` prevents transforming when importing `~/Sidebar`. But then when babel transforms `Sidebar.tsx` it imports from `./Button` and that does attempt a transform which then crashes because it's not a directory. I obviously can't block everything under `./` because that would block all local imports... So what to do? I'll try to make a test for that.

**Comment by jmeistrich** [comment]
Regarding the `guessAtPath` issue: I've made a minimal repro now and the main issue seems to be relative paths, such as `import Component from ./Component`. I can't seem to reproduce the problem within this library's tests (it doesn't traverse into files other than code.tsx?), but it reproduces in a simple Next.js project that I made: https://github.com/jmeistrich/tailwindcss-react-native-next

You can run it with `npm run dev`

You should hit the `guessAtPath` crash when importing DOMComponent. It crashes in two instances (search for DomComponentFAIL) when it is imported as a relative path. From my testing it looks like `require.resolve` doesn't seem to be working on relative paths? Or am I just still configuring it wrong? Is there some way I'm missing that won't crash on relative paths? Configuring `./` to either allow or block isn't really workable because that could be anywhere.

I think solving number 1 above could effectively take care of this (and be a big performance boost). Because the "web" folder is contained in the tailwind config, it's trying to process all the files in there, and hitting this error while processing all the imports. Can the `allowModules` and `blockModules` configs be used to block processing the files too, rather than just imports? Or can there be another config to set `allowRelativeModules`?

I don't think the tailwind config should necessarily be the unchangeable source of truth (on web at least), since it is probably fairly common to do as we're doing and have tailwind processing both React and RN components. Some additional config to override which paths the babel plugin should process would be great.

**Comment by marklawlor** [comment]
Thanks for that repo, I believe this is fixed in `0.0.30`


![Screenshot from 2022-04-23 10-06-12](https://user-images.githubusercontent.com/3946701/164836411-9eb9e511-2262-48e9-b2e1-52813b973035.png)


You can update your `babel.config.js` to

```
'tailwindcss-react-native/babel',
{
    platform: 'web',
    allowModules: ['rn/*'],
},
```

Allow modules now correctly only allows those modules, so there is no need for `blockModules`.

I'm still not 100% convinced that the code around exotic imports is perfect, so let me know if you have issues.

**Comment by jmeistrich** [comment]
Yes, this looks like it works! But FYI I did need to add `react-native` to `allowModules`. And in our real app I also had to add `react-native-web` (I'm not sure why it's different). Those should probably be allowed implicitly?
